library/src/main/java/org/xmtp/android/library/Crypto.kt

```diff
@@ -0,0 +1,75 @@
+package org.xmtp.android.library
+
+import com.google.crypto.tink.subtle.Hkdf
+import com.google.protobuf.kotlin.toByteString
+import org.xmtp.proto.message.contents.CiphertextOuterClass
+import java.security.GeneralSecurityException
+import java.security.SecureRandom
+import javax.crypto.Cipher
+import javax.crypto.spec.GCMParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+typealias CipherText = CiphertextOuterClass.Ciphertext
+
+class Crypto {
+    companion object {
+        fun encrypt(
+            secret: ByteArray,
+            message: ByteArray,
+            additionalData: ByteArray = byteArrayOf()
+        ): CipherText? {
+            return try {
+                val salt = SecureRandom().generateSeed(32)
+                val nonceData = SecureRandom().generateSeed(12)
+                val cipher = Cipher.getInstance("AES/GCM/NoPadding")
+
+                val key = Hkdf.computeHkdf("HMACSHA256", secret, salt, null, 32)
+                val keySpec = SecretKeySpec(key, "AES")
+                val gcmSpec = GCMParameterSpec(128, nonceData)
+
+                cipher.init(Cipher.ENCRYPT_MODE, keySpec, gcmSpec)
+                if (additionalData.isNotEmpty()) {
+                    cipher.updateAAD(additionalData)
+                }
+                val final = cipher.doFinal(message)
+
+                CiphertextOuterClass.Ciphertext.newBuilder().apply {
+                    aes256GcmHkdfSha256Builder.apply {
+                        payload = final.toByteString()
+                        hkdfSalt = salt.toByteString()
+                        gcmNonce = nonceData.toByteString()
+                    }.build()
+                }.build()
+            } catch (e: GeneralSecurityException) {
+                e.printStackTrace()
```

Comment:
It's likely worth removing `e.printStackTrace()`s in favor of Log.e(). We also probably want to strip log statements from the published library to avoid leaking any key info during obfuscation. Something like this can be added to the proguard file so they're still visible while in debug mode but not in any consumers code:

```
-assumenosideeffects class android.util.Log {
    public static boolean isLoggable(java.lang.String, int);
    public static int v(...);
    public static int w(...);
    public static int e(...);
    public static int d(...);
    public static int i(...);
}
```